addon/components/my-materials.js

```diff
@@ -1,61 +1,76 @@
-import Component from '@ember/component';
-import { computed } from '@ember/object';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
 import { isPresent } from '@ember/utils';
-import SortableTable from 'ilios-common/mixins/sortable-table';
-import { task, timeout } from 'ember-concurrency';
+import { timeout } from 'ember-concurrency';
+import {restartableTask} from "ember-concurrency-decorators";
 
 const DEBOUNCE_DELAY = 250;
 
-export default Component.extend(SortableTable, {
-  classNames: ['my-materials'],
+export default class MyMaterials extends Component {
+  @tracked materials =  [];
+  @tracked loading = true;
 
-  courseIdFilter: null,
-  filter: null,
-  materials: null,
-  sortBy: null,
-  setCourseIdFilter() {},
-  setFilter() {},
-  setSortBy() {},
-
-  filteredMaterials: computed('courseIdFilter', 'filter', 'materials.[]', async function() {
-    const courseIdFilter = this.courseIdFilter;
-    const filter = this.filter;
-    let materials = await this.materials;
+  get filteredMaterials() {
+    let materials = this.materials;
+    if (! this.materials) {
+      return [];
+    }
 
-    if (isPresent(courseIdFilter)) {
-      materials = materials.filterBy('course', courseIdFilter);
+    if (isPresent(this.args.courseIdFilter)) {
+      materials = this.materials.filterBy('course', this.args.courseIdFilter);
     }
 
-    if (isPresent(filter)) {
+    if (isPresent(this.args.filter)) {
       materials = materials.filter(({ courseTitle, instructors, sessionTitle, title }) => {
         let searchString = `${title} ${courseTitle} ${sessionTitle} `;
 
         if (isPresent(instructors)) {
           searchString += instructors.join(' ');
         }
 
-        return searchString.toLowerCase().includes(filter.toLowerCase());
+        return searchString.toLowerCase().includes(this.args.filter.toLowerCase());
       });
     }
-
     return materials;
-  }),
+  }
 
-  courses: computed('materials.[]', async function() {
-    const materials = await this.materials;
-    return materials.map((material) => {
+  get courses() {
+    if (! this.materials) {
+      return [];
+    }
+    return this.materials.map((material) => {
       return { id: material.course, title: material.courseTitle };
     }).uniqBy('id').sortBy('title');
-  }),
+  }
 
-  actions: {
-    sortString(a, b){
-      return a.localeCompare(b);
+  get sortedAscending() {
+    return this.args.sortBy.search(/desc/) === -1;
+  }
+
+  @action
+  sortString(a, b){
+    return a.localeCompare(b);
+  }
+
+  @action
+  sortBy(what){
+    if (this.args.sortBy === what){
+      what += ':desc';
     }
-  },
+    this.args.setSortBy(what);
+  }
+
+  @restartableTask
+  *load(element, [materials]) {
+    this.loading = true;
+    this.materials = yield materials;
```

Comment:
Since materials are passed in as a resolved promise `@materials={{await this.materials}}` this `yield` seems unnecessary.

Reply:
done. simplified.